### PR TITLE
1.7/1408 extended rules

### PR DESCRIPTION
**Pull request by charlesstrange2**
Add two rules to detect children being over school and secondary school age
Add them to the rulesset
Add tests for them
Make factories for children that fit that.
Change the language on the school and school age to be more friendly

### CODE REVIEW

File: app/Services/VoucherEvaluator/Evaluations/ChildIsSchoolAge.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+{
+    const REASON = 'school age (ineligible)';
+    private $specification;
+
+    public function __construct(Carbon $offsetDate = null, $value = 3)
+    {
+        parent::__construct($offsetDate, $value);
+
+        $this->specification = new AndSpec(
+            new IsBorn(),
```
Comment (Hoolean) [inline app/Services/VoucherEvaluator/Evaluations/ChildIsSchoolAge.php]: I think this is redundant - if a child is over school age they're also born - but not doing any harm.

File: app/Services/VoucherEvaluator/Evaluations/ChildIsSecondarySchoolAge.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+{
+    const REASON = 'secondary school age (ineligible)';
+    private $specification;
+
+    public function __construct(Carbon $offsetDate = null, $value = 3)
+    {
+        parent::__construct($offsetDate, $value);
+
+        $this->specification = new AndSpec(
+            new IsBorn(),
```
Comment (Hoolean) [inline app/Services/VoucherEvaluator/Evaluations/ChildIsSecondarySchoolAge.php]: Same redundancy as above.

File: tests/Unit/Services/VoucherEvaluator/VoucherEvaluatorTest.php [hunk 3]
```diff
@@ -176,6 +197,27 @@ public function itNoticesWhenAChildIsAlmostSchoolAge()
         $this->assertContains(self::NOTICE_TYPES['ChildIsAlmostSchoolAge'], $notices);
     }
 
+    /** @test */
+    public function itNoticesWhenAChildIsAlmostSecondarySchoolAge()
+    {
+        // Need to change the values we use for school start to next month's integer
+        Config::set('arc.school_month', Carbon::now()->addMonth(1)->month);
```
Comment (Hoolean) [inline tests/Unit/Services/VoucherEvaluator/VoucherEvaluatorTest.php]: If it is December and this month wraps around to January, this test will not behave as expected.

File: app/Services/VoucherEvaluator/Evaluations/ChildIsSchoolAge.php (hunk 1 above)
Comment (charlesstrange2) [reply app/Services/VoucherEvaluator/Evaluations/ChildIsSchoolAge.php]: In this rule, that's true; we're never going to have a 5 year pregnancy. I'll mark it for removal at some point. In other rules it helps check for overdue pregnancies.

File: tests/Unit/Services/VoucherEvaluator/VoucherEvaluatorTest.php (hunk 3 above)
Comment (charlesstrange2) [reply tests/Unit/Services/VoucherEvaluator/VoucherEvaluatorTest.php]: As discussed, Card and put in tech debt to be dealt with next heartbeat.

**Pull request merged**